cachito/workers/utils.py

```diff
@@ -19,25 +24,42 @@ def extract_app_src(archive_path, parent_dir):
     return os.path.join(parent_dir, 'app')
 
 
-def get_request_bundle_dir(request_id):
+class RequestBundleDir(paths.RequestBundleDir):
     """
-    Get the path to the directory where the bundle is being created for the request.
-
-    :param int request_id: the request ID to get the directory for
-    :return: the path to the where the bundle is being created
-    :rtype: str
+    Represents a concrete request bundle directory used on worker whose root
+    directory defaults to ``cachito_bundles_dir`` in config.
     """
-    config = get_worker_config()
-    return os.path.join(config.cachito_bundles_dir, 'temp', str(request_id))
 
+    def __new__(cls, request_id):
+        root_dir = get_worker_config().cachito_bundles_dir
+        self = super().__new__(cls, request_id, root_dir)
+
+        log.debug('Ensure directory %s exists.', self)
+        self.mkdir(parents=True, exist_ok=True)
+
+        log.debug('Ensure directory %s exists.', self.deps_dir)
+        self.deps_dir.mkdir(parents=True, exist_ok=True)
+
+        return self
 
-def get_request_bundle_path(request_id):
+
+class SourcesDir(type(pathlib.Path())):
     """
-    Get the path to the request's bundle.
+    Represents a sources directory tree for a package
```

Comment:
Could you add a note here that says that the directories are created when this is instantiated?